tests/e2e/vmlistener_test.go

```diff
@@ -127,29 +156,49 @@ func (s *VMListenerTestParams) TestBasicFailover(c *C) {
 	esxutil.PowerOnVM(s.vm1Name)
 	isStatusChanged := esxutil.WaitForExpectedState(esxutil.GetVMPowerState, s.vm1Name, properties.PowerOnState)
 	c.Assert(isStatusChanged, Equals, true, Commentf("VM [%s] should be powered on state", s.vm1Name))
+	c.Assert(verification.IsVDVSIsRunning(s.vm1), Equals, true, Commentf("vDVS is not running on [%s]", s.vm1Name))
 
 	// Status should be detached
-	status := verification.VerifyDetachedStatus(s.volumeName, s.vm1, s.esx)
+	status = verification.VerifyDetachedStatus(s.volumeName, s.vm1, s.esx)
 	c.Assert(status, Equals, true, Commentf("Volume %s is still attached", s.volumeName))
 
+	// Remove the volume
```

Comment:
Yes, I have considered the same. However, this is not a must, right? Even if the container exists after VM restarts, the volume should still be in detached status, correct? So removing volume should not run into issue here.

I think both are acceptable. I keep it in TearDown just to avoid a little bit code duplication. Comments?